extensions-core/kinesis-indexing-service/src/test/java/org/apache/druid/indexing/kinesis/KinesisRecordSupplierTest.java

```diff
@@ -410,6 +414,53 @@ public void testPollWithKinesisInternalFailure() throws InterruptedException
     Assert.assertEquals(SHARDS_LAG_MILLIS, recordSupplier.getPartitionResourcesTimeLag());
   }
 
+  @Test
+  public void testRecoverableException_IOException()
+  {
+    Assert.assertTrue(KinesisRecordSupplier.isClientExceptionRecoverable(new AmazonClientException(new IOException())));
+  }
+
+  @Test
+  public void testRecoverableException_RequestTimeout()
+  {
+    AmazonServiceException ex = new AmazonServiceException(null);
+    ex.setErrorCode("RequestTimeout");
+    Assert.assertTrue(KinesisRecordSupplier.isClientExceptionRecoverable(ex));
+  }
+
+  @Test
+  public void testRecoverableException_500()
+  {
+    AmazonServiceException ex = new AmazonServiceException(null);
+    ex.setStatusCode(500);
+    Assert.assertTrue(KinesisRecordSupplier.isClientExceptionRecoverable(ex));
+  }
+
+  @Test
+  public void testRecoverableException_502()
+  {
+    AmazonServiceException ex = new AmazonServiceException(null);
+    ex.setStatusCode(502);
+    Assert.assertTrue(KinesisRecordSupplier.isClientExceptionRecoverable(ex));
+  }
+
+  @Test
+  public void testRecoverableException_503()
+  {
+    AmazonServiceException ex = new AmazonServiceException(null);
+    ex.setStatusCode(503);
+    Assert.assertTrue(KinesisRecordSupplier.isClientExceptionRecoverable(ex));
+  }
+
+  @Test
+  public void testRecoverableException_ProvisionedThroughputExceededException()
+  {
+    AmazonServiceException ex = new AmazonServiceException(null);
+    ex.setErrorCode("ProvisionedThroughputExceededException");
+    Assert.assertTrue(KinesisRecordSupplier.isClientExceptionRecoverable(ex));
+  }
+
+
```

Comment:
Could you add tests for `ClockSkewError` and non-recoverable exceptions as well please